django_bmemcached/memcached.py

```diff
@@ -1,14 +1,44 @@
 import os
+
+import bmemcached
 from django.core.cache.backends import memcached
 
+# keys that are acceptable in the Django
+# settings.CACHES[...]["OPTIONS"] dictionary.
+VALID_CACHE_OPTIONS = (
+    "username",
+    "password",
+    "compression",
+    "socket_timeout",
+    "pickler",
+    "unpickler",
+    "pickle_protocol",
+)
+
+
+class InvalidCacheOptions(Exception):
+    """Custom error raised when Client initialisation fails."""
```

Comment:
This error, when raised, will output a message that lists the bad options passed, and the list of valid options:

```
django_bmemcached.memcached.InvalidCacheOptions: Error initialising BMemcached client - invalid options: bad_option.
Valid options: username, password, compression, socket_timeout, pickler, unpickler, pickle_protocol.
Please check your Django cache 'OPTIONS'.
```

Reply:
I guess it is ok, but from the user perspective upgrading a minor version introducing some break could be a problem, what do you think?

Reply:
My personal preference (but it's not my lib ;-)) is to fail hard on something that is a pure config / settings issue, as it's a binary problem - the config is valid, or invalid, and the remedial work required is very simple (and non-functional). Once it's fixed the problem disappears. In practice raising this issue will fail for everyone (ourselves included) that their current config is invalid, which is something they should probably be aware of?

If I saw a release note that said "will now raise an exception if config options are invalid" I wouldn't have an issue. YMMV.

Reply:
Ok, let's leave with the exception then